src/views/TimeOutWrapper/index.tsx

```diff
@@ -1,38 +1,152 @@
     (state: AppState) => state.auth.lastActiveAt
   );
 
-  const { authService }: { authService: any } = useOktaAuth();
-  registerExpire(authService, lastActiveAt);
+  const oneSecond = Duration.fromObject({ seconds: 1 }).as('milliseconds');
+  const fiveMinutes = Duration.fromObject({ minutes: 5 }).as('milliseconds');
+
+  const registerExpire = async () => {
+    const tokenManager = await authService.getTokenManager();
+    // clear the old listener so we don't register millions of them
+    tokenManager.off('expired');
+    tokenManager.on('expired', (key: any) => {
+      const activeSessionWindow = DateTime.local()
+        .minus({ minutes: 14 })
+        .toMillis();
+      if (lastActiveAt > activeSessionWindow) {
+        tokenManager.renew(key);
+      } else {
+        authService.logout('/login');
+      }
+    });
+  };
+
+  const formatSessionTimeRemaining = (
+    timeRemaining: number
+  ): [number, string] => {
+    const seconds = Math.floor(
+      Duration.fromMillis(timeRemaining).as('seconds')
+    );
+
+    const minutes = seconds / 60;
+    // Using Math.ceil() for greater than one minute
+    // 299 seconds = 4.983 minutes, but should still display 5 minutes
+    // Using Math.floor() for less than one minute
+    // 59 seconds = .983 minutes, Using floor so minutes is 0 to display 59 secounds
+    const wholeMinutes = minutes > 1 ? Math.ceil(minutes) : Math.floor(minutes);
+
+    return wholeMinutes > 0
+      ? [wholeMinutes, 'minute']
+      : [Math.floor(seconds), 'second'];
+  };
+
+  const handleModalExit = async () => {
+    setIsModalOpen(false);
+    const tokenManager = await authService.getTokenManager();
+    await tokenManager.renew('accessToken');
+    await tokenManager.renew('idToken');
+    dispatch(updateLastActiveAt);
+  };
+
+  useInterval(
+    async () => {
+      const currentTime = Date.now();
+      const tokenManager = await authService.getTokenManager();
+      const accessToken = await tokenManager.get('accessToken');
+      if (accessToken) {
+        const tokenExp = DateTime.fromSeconds(accessToken.expiresAt).toMillis();
+
+        if (tokenExp - currentTime >= 0) {
+          setTimeRemainingArr(
+            formatSessionTimeRemaining(tokenExp - currentTime)
+          );
+        }
+      }
+    },
+    isModalOpen ? oneSecond : null
+  );
+
+  useInterval(
+    async () => {
+      const currentTime = Date.now();
+      const tokenManager = await authService.getTokenManager();
+      const accessToken = await tokenManager.get('accessToken');
+      const tokenExp = DateTime.fromSeconds(accessToken.expiresAt).toMillis();
+
+      if (tokenExp - currentTime < fiveMinutes) {
+        setTimeRemainingArr(formatSessionTimeRemaining(tokenExp - currentTime));
+        setIsModalOpen(true);
+      }
+    },
+    authState.isAuthenticated && !isModalOpen ? oneSecond : null
+  );
+
+  useEffect(() => {
+    if (authState.isAuthenticated) {
+      registerExpire();
```

Comment:
Throwing this in a hook is rad! We might want to poke and see how many times the listener is being set. We could maybe edit some of the code in that function, but we can do that as a follow-on story.